Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,311 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers;
+
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Connection\AbstractConnection;
+use PhpAmqpLib\Connection\AMQPStreamConnection;
+use PhpAmqpLib\Exception\AMQPIOException;
+use PhpAmqpLib\Message\AMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Consumers\AbstractAsyncConsumer;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class PhpAmqpLibDriver.
+ */
+class PhpAmqpLibDriver extends Service implements AsyncQueueDriverInterface
+{
+    use UsesSerializer;
+
+    /**
+     * This field specifies the prefetch window size in octets.
+     * The server will send a message in advance if it is equal to or smaller in size than the available prefetch size
+     * (and also falls into other prefetch limits).
+     * May be set to zero, meaning "no specific limit", although other prefetch limits may still apply.
+     * The prefetch-size is ignored if the no-ack option is set.
+     */
+    const PREFETCH_SIZE = null;
+
+    /**
+     * Represents the amount of message to consume by iteration
+     */
+    const PREFETCH_COUNT = 10;
+
+    /*
+     * Default exchange name
+     */
+    const EXCHANGE_NAME = '';
+
+    /**
+     * Default port to connection
+     */
+    const DEFAULT_PORT = 5672;
+
+    /**
+     * Default host to connect
+     */
+    const DEFAULT_HOST = 'localhost';
+
+    /**
+     * @var string
+     */
+    protected $format = AsyncQueueDriverInterface::FORMAT_JSON;
+
+    /**
+     * @var AMQPChannel|null
+     */
+    private $channel;
+
+    /**
+     * @var \AMQPExchange|null
+     */
+    private $exchange;
+
+    /**
+     * @var array|null
+     */
+    protected $connectionsData;
+
+    /**
+     * @var AMQPStreamConnection
+     */
+    protected $stream;
+
+    public function __destruct()
+    {
+        $this->disconnect();
+    }
+
+    /**
+     * Method responsible to catch the parameters to configure the driver.
```

Comment:
This could be deleted because we are already using inherit doc

Reply:
Done.